stock-tracker/src/socket/subscriptions.js

```diff
@@ -3,10 +3,15 @@ import { eventActions } from "./eventActions";
 
 export const createSocketSubscriptions = dispatch => {
   socketService.create();
-  const unsubscribeSockets = eventActions.map(([event, actionCreator]) =>
-    socketService.createSocketSubscription(event, payload =>
-      dispatch(actionCreator(payload))
-    )
+  const unsubscribeSockets = eventActions.map(
+    ([event, actionCreator, errorActionCreator]) =>
+      socketService.createSocketSubscription(event, payload => {
+        if (payload.data) {
```

Comment:
major: I'd recommend checking `if (payload.isError)`, rather than `payload.data`.
Currently, if `payload.data` is `falsey`, it would throw an error action! There's no reason why `payload.data` can't be `0`, `false`, `""`, or anything else `falsey`.

Reply:
Did not think about this thanks for spotting this out, changed it now.